src/Handler/SentryRavenHandler.php

```diff
@@ -80,10 +80,10 @@ protected function write(array $record)
             $options['extra']['extra'] = $record['extra'];
         }
 
-        // New for phptek/sentry
+        // Save stack
         $stack = false;
-        if (!empty($record['stack'])) {
-            $stack = (bool) $record['stack'];
+        if (!empty($record['stacktrace'])) {
+            $stack = $record['stacktrace'];
```

Comment:
Use actual trace array instead of bool to prevent sentry re-generating trace incorrectly.

Reply:
I dislike multi-type parameters intensely. If `Raven::captureMessage()` wants to accept an empty param for its 4th arg, it should allow an empty array....but then it goes on to accept `$stack` as `null` in `Raven::capture()` so what can you do...

I don't recall specifically why I cast to a `bool`, but it could be related to Raven_Client:909:

```
...
        if ((!$stack && $this->auto_log_stacks) || $stack === true) {
            $stack = debug_backtrace();
...
```

Reply:
If you cast it as a bool, it forces the stacktrace to be generated internally.

If you pass an array, it uses that value as the actual stacktrace.

The problem is that monolog stores the stacktrace for later error reporting (register_shutdown_function), so it's useless to try to use `debug_backtrace` in the capture since you are no longer in the stack.

I think false is just `no stacktrace`.